Traceback (most recent call last):
  File "C:\Users\janko\AppData\Local\Programs\Python\Python312\Lib\site-packages\jupyter_cache\executors\utils.py", line 58, in single_nb_execution
    executenb(
  File "C:\Users\janko\AppData\Local\Programs\Python\Python312\Lib\site-packages\nbclient\client.py", line 1319, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\janko\AppData\Local\Programs\Python\Python312\Lib\site-packages\jupyter_core\utils\__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\janko\AppData\Local\Programs\Python\Python312\Lib\asyncio\base_events.py", line 687, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "C:\Users\janko\AppData\Local\Programs\Python\Python312\Lib\site-packages\nbclient\client.py", line 709, in async_execute
    await self.async_execute_cell(
  File "C:\Users\janko\AppData\Local\Programs\Python\Python312\Lib\site-packages\nbclient\client.py", line 1062, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "C:\Users\janko\AppData\Local\Programs\Python\Python312\Lib\site-packages\nbclient\client.py", line 918, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
import pytube
yt = pytube.YouTube('https://www.youtube.com/watch?v=P4IrAXbpeyI') # kreiramo instanco objekta (kaj to točno pomeni, spoznamo pozneje)
yt.streams.get_highest_resolution().download() #Prenesemo video
------------------


[1;31m---------------------------------------------------------------------------[0m
[1;31mHTTPError[0m                                 Traceback (most recent call last)
Cell [1;32mIn[61], line 3[0m
[0;32m      1[0m [38;5;28;01mimport[39;00m [38;5;21;01mpytube[39;00m
[0;32m      2[0m yt [38;5;241m=[39m pytube[38;5;241m.[39mYouTube([38;5;124m'[39m[38;5;124mhttps://www.youtube.com/watch?v=P4IrAXbpeyI[39m[38;5;124m'[39m) [38;5;66;03m# kreiramo instanco objekta (kaj to točno pomeni, spoznamo pozneje)[39;00m
[1;32m----> 3[0m [43myt[49m[38;5;241;43m.[39;49m[43mstreams[49m[38;5;241m.[39mget_highest_resolution()[38;5;241m.[39mdownload() [38;5;66;03m#Prenesemo video[39;00m

File [1;32m~\AppData\Local\Programs\Python\Python312\Lib\site-packages\pytube\__main__.py:296[0m, in [0;36mYouTube.streams[1;34m(self)[0m
[0;32m    291[0m [38;5;250m[39m[38;5;124;03m"""Interface to query both adaptive (DASH) and progressive streams.[39;00m
[0;32m    292[0m 
[0;32m    293[0m [38;5;124;03m:rtype: :class:`StreamQuery <StreamQuery>`.[39;00m
[0;32m    294[0m [38;5;124;03m"""[39;00m
[0;32m    295[0m [38;5;28mself[39m[38;5;241m.[39mcheck_availability()
[1;32m--> 296[0m [38;5;28;01mreturn[39;00m StreamQuery([38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mfmt_streams[49m)

File [1;32m~\AppData\Local\Programs\Python\Python312\Lib\site-packages\pytube\__main__.py:176[0m, in [0;36mYouTube.fmt_streams[1;34m(self)[0m
[0;32m    172[0m     [38;5;28;01mreturn[39;00m [38;5;28mself[39m[38;5;241m.[39m_fmt_streams
[0;32m    174[0m [38;5;28mself[39m[38;5;241m.[39m_fmt_streams [38;5;241m=[39m []
[1;32m--> 176[0m stream_manifest [38;5;241m=[39m extract[38;5;241m.[39mapply_descrambler([38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mstreaming_data[49m)
[0;32m    178[0m [38;5;66;03m# If the cached js doesn't work, try fetching a new js file[39;00m
[0;32m    179[0m [38;5;66;03m# https://github.com/pytube/pytube/issues/1054[39;00m
[0;32m    180[0m [38;5;28;01mtry[39;00m:

File [1;32m~\AppData\Local\Programs\Python\Python312\Lib\site-packages\pytube\__main__.py:157[0m, in [0;36mYouTube.streaming_data[1;34m(self)[0m
[0;32m    154[0m [38;5;129m@property[39m
[0;32m    155[0m [38;5;28;01mdef[39;00m [38;5;21mstreaming_data[39m([38;5;28mself[39m):
[0;32m    156[0m [38;5;250m    [39m[38;5;124;03m"""Return streamingData from video info."""[39;00m
[1;32m--> 157[0m     [38;5;28;01mif[39;00m [38;5;124m'[39m[38;5;124mstreamingData[39m[38;5;124m'[39m [38;5;129;01min[39;00m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mvid_info[49m:
[0;32m    158[0m         [38;5;28;01mreturn[39;00m [38;5;28mself[39m[38;5;241m.[39mvid_info[[38;5;124m'[39m[38;5;124mstreamingData[39m[38;5;124m'[39m]
[0;32m    159[0m     [38;5;28;01melse[39;00m:

File [1;32m~\AppData\Local\Programs\Python\Python312\Lib\site-packages\pytube\__main__.py:246[0m, in [0;36mYouTube.vid_info[1;34m(self)[0m
[0;32m    242[0m     [38;5;28;01mreturn[39;00m [38;5;28mself[39m[38;5;241m.[39m_vid_info
[0;32m    244[0m innertube [38;5;241m=[39m InnerTube(use_oauth[38;5;241m=[39m[38;5;28mself[39m[38;5;241m.[39muse_oauth, allow_cache[38;5;241m=[39m[38;5;28mself[39m[38;5;241m.[39mallow_oauth_cache)
[1;32m--> 246[0m innertube_response [38;5;241m=[39m [43minnertube[49m[38;5;241;43m.[39;49m[43mplayer[49m[43m([49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mvideo_id[49m[43m)[49m
[0;32m    247[0m [38;5;28mself[39m[38;5;241m.[39m_vid_info [38;5;241m=[39m innertube_response
[0;32m    248[0m [38;5;28;01mreturn[39;00m [38;5;28mself[39m[38;5;241m.[39m_vid_info

File [1;32m~\AppData\Local\Programs\Python\Python312\Lib\site-packages\pytube\innertube.py:448[0m, in [0;36mInnerTube.player[1;34m(self, video_id)[0m
[0;32m    444[0m query [38;5;241m=[39m {
[0;32m    445[0m     [38;5;124m'[39m[38;5;124mvideoId[39m[38;5;124m'[39m: video_id,
[0;32m    446[0m }
[0;32m    447[0m query[38;5;241m.[39mupdate([38;5;28mself[39m[38;5;241m.[39mbase_params)
[1;32m--> 448[0m [38;5;28;01mreturn[39;00m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_call_api[49m[43m([49m[43mendpoint[49m[43m,[49m[43m [49m[43mquery[49m[43m,[49m[43m [49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mbase_data[49m[43m)[49m

File [1;32m~\AppData\Local\Programs\Python\Python312\Lib\site-packages\pytube\innertube.py:390[0m, in [0;36mInnerTube._call_api[1;34m(self, endpoint, query, data)[0m
[0;32m    386[0m         headers[[38;5;124m'[39m[38;5;124mAuthorization[39m[38;5;124m'[39m] [38;5;241m=[39m [38;5;124mf[39m[38;5;124m'[39m[38;5;124mBearer [39m[38;5;132;01m{[39;00m[38;5;28mself[39m[38;5;241m.[39maccess_token[38;5;132;01m}[39;00m[38;5;124m'[39m
[0;32m    388[0m headers[38;5;241m.[39mupdate([38;5;28mself[39m[38;5;241m.[39mheader)
[1;32m--> 390[0m response [38;5;241m=[39m [43mrequest[49m[38;5;241;43m.[39;49m[43m_execute_request[49m[43m([49m
[0;32m    391[0m [43m    [49m[43mendpoint_url[49m[43m,[49m
[0;32m    392[0m [43m    [49m[38;5;124;43m'[39;49m[38;5;124;43mPOST[39;49m[38;5;124;43m'[39;49m[43m,[49m
[0;32m    393[0m [43m    [49m[43mheaders[49m[38;5;241;43m=[39;49m[43mheaders[49m[43m,[49m
[0;32m    394[0m [43m    [49m[43mdata[49m[38;5;241;43m=[39;49m[43mdata[49m
[0;32m    395[0m [43m[49m[43m)[49m
[0;32m    396[0m [38;5;28;01mreturn[39;00m json[38;5;241m.[39mloads(response[38;5;241m.[39mread())

File [1;32m~\AppData\Local\Programs\Python\Python312\Lib\site-packages\pytube\request.py:37[0m, in [0;36m_execute_request[1;34m(url, method, headers, data, timeout)[0m
[0;32m     35[0m [38;5;28;01melse[39;00m:
[0;32m     36[0m     [38;5;28;01mraise[39;00m [38;5;167;01mValueError[39;00m([38;5;124m"[39m[38;5;124mInvalid URL[39m[38;5;124m"[39m)
[1;32m---> 37[0m [38;5;28;01mreturn[39;00m [43murlopen[49m[43m([49m[43mrequest[49m[43m,[49m[43m [49m[43mtimeout[49m[38;5;241;43m=[39;49m[43mtimeout[49m[43m)[49m

File [1;32m~\AppData\Local\Programs\Python\Python312\Lib\urllib\request.py:215[0m, in [0;36murlopen[1;34m(url, data, timeout, cafile, capath, cadefault, context)[0m
[0;32m    213[0m [38;5;28;01melse[39;00m:
[0;32m    214[0m     opener [38;5;241m=[39m _opener
[1;32m--> 215[0m [38;5;28;01mreturn[39;00m [43mopener[49m[38;5;241;43m.[39;49m[43mopen[49m[43m([49m[43murl[49m[43m,[49m[43m [49m[43mdata[49m[43m,[49m[43m [49m[43mtimeout[49m[43m)[49m

File [1;32m~\AppData\Local\Programs\Python\Python312\Lib\urllib\request.py:521[0m, in [0;36mOpenerDirector.open[1;34m(self, fullurl, data, timeout)[0m
[0;32m    519[0m [38;5;28;01mfor[39;00m processor [38;5;129;01min[39;00m [38;5;28mself[39m[38;5;241m.[39mprocess_response[38;5;241m.[39mget(protocol, []):
[0;32m    520[0m     meth [38;5;241m=[39m [38;5;28mgetattr[39m(processor, meth_name)
[1;32m--> 521[0m     response [38;5;241m=[39m [43mmeth[49m[43m([49m[43mreq[49m[43m,[49m[43m [49m[43mresponse[49m[43m)[49m
[0;32m    523[0m [38;5;28;01mreturn[39;00m response

File [1;32m~\AppData\Local\Programs\Python\Python312\Lib\urllib\request.py:630[0m, in [0;36mHTTPErrorProcessor.http_response[1;34m(self, request, response)[0m
[0;32m    627[0m [38;5;66;03m# According to RFC 2616, "2xx" code indicates that the client's[39;00m
[0;32m    628[0m [38;5;66;03m# request was successfully received, understood, and accepted.[39;00m
[0;32m    629[0m [38;5;28;01mif[39;00m [38;5;129;01mnot[39;00m ([38;5;241m200[39m [38;5;241m<[39m[38;5;241m=[39m code [38;5;241m<[39m [38;5;241m300[39m):
[1;32m--> 630[0m     response [38;5;241m=[39m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mparent[49m[38;5;241;43m.[39;49m[43merror[49m[43m([49m
[0;32m    631[0m [43m        [49m[38;5;124;43m'[39;49m[38;5;124;43mhttp[39;49m[38;5;124;43m'[39;49m[43m,[49m[43m [49m[43mrequest[49m[43m,[49m[43m [49m[43mresponse[49m[43m,[49m[43m [49m[43mcode[49m[43m,[49m[43m [49m[43mmsg[49m[43m,[49m[43m [49m[43mhdrs[49m[43m)[49m
[0;32m    633[0m [38;5;28;01mreturn[39;00m response

File [1;32m~\AppData\Local\Programs\Python\Python312\Lib\urllib\request.py:559[0m, in [0;36mOpenerDirector.error[1;34m(self, proto, *args)[0m
[0;32m    557[0m [38;5;28;01mif[39;00m http_err:
[0;32m    558[0m     args [38;5;241m=[39m ([38;5;28mdict[39m, [38;5;124m'[39m[38;5;124mdefault[39m[38;5;124m'[39m, [38;5;124m'[39m[38;5;124mhttp_error_default[39m[38;5;124m'[39m) [38;5;241m+[39m orig_args
[1;32m--> 559[0m     [38;5;28;01mreturn[39;00m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_call_chain[49m[43m([49m[38;5;241;43m*[39;49m[43margs[49m[43m)[49m

File [1;32m~\AppData\Local\Programs\Python\Python312\Lib\urllib\request.py:492[0m, in [0;36mOpenerDirector._call_chain[1;34m(self, chain, kind, meth_name, *args)[0m
[0;32m    490[0m [38;5;28;01mfor[39;00m handler [38;5;129;01min[39;00m handlers:
[0;32m    491[0m     func [38;5;241m=[39m [38;5;28mgetattr[39m(handler, meth_name)
[1;32m--> 492[0m     result [38;5;241m=[39m [43mfunc[49m[43m([49m[38;5;241;43m*[39;49m[43margs[49m[43m)[49m
[0;32m    493[0m     [38;5;28;01mif[39;00m result [38;5;129;01mis[39;00m [38;5;129;01mnot[39;00m [38;5;28;01mNone[39;00m:
[0;32m    494[0m         [38;5;28;01mreturn[39;00m result

File [1;32m~\AppData\Local\Programs\Python\Python312\Lib\urllib\request.py:639[0m, in [0;36mHTTPDefaultErrorHandler.http_error_default[1;34m(self, req, fp, code, msg, hdrs)[0m
[0;32m    638[0m [38;5;28;01mdef[39;00m [38;5;21mhttp_error_default[39m([38;5;28mself[39m, req, fp, code, msg, hdrs):
[1;32m--> 639[0m     [38;5;28;01mraise[39;00m HTTPError(req[38;5;241m.[39mfull_url, code, msg, hdrs, fp)

[1;31mHTTPError[0m: HTTP Error 400: Bad Request

